source/grid/grid_generator.cc

```diff
@@ -8440,7 +8459,8 @@ namespace GridGenerator
 
         // temporary array storing the global indices of each cell entity in the
         // sequence: vertices, edges/faces, cell
-        std::array<unsigned int, 4> local_vertex_indices;
+        std::array<unsigned int, 5> local_vertex_indices;
```

Comment:
You should be able to replace the 5 with the expression that is either 5 or 4 depending on dimension:

```suggestion
        std::array<unsigned int, (dim==3)?5:4> local_vertex_indices;
```

Then you don't need to introduce the extra variable.